src/org/traccar/notification/NotificationFormatter.java

```diff
@@ -15,223 +15,65 @@
-                        position.getLatitude(), position.getLongitude(),
-                        Context.getGeofenceManager().getGeofence(event.getGeofenceId()).getName());
-                break;
-            case Event.TYPE_ALARM:
-                formatter.format(MESSAGE_TEMPLATE_TYPE_ALARM, device.getName(), event.getServerTime(),
-                        position.getLatitude(), position.getLongitude(),
-                        position.getAttributes().get(Position.KEY_ALARM));
-                break;
-            case Event.TYPE_IGNITION_ON:
-                formatter.format(MESSAGE_TEMPLATE_TYPE_IGNITION_ON, device.getName(), position.getFixTime(),
-                        position.getLatitude(), position.getLongitude());
-                break;
-            case Event.TYPE_IGNITION_OFF:
-                formatter.format(MESSAGE_TEMPLATE_TYPE_IGNITION_OFF, device.getName(), position.getFixTime(),
-                        position.getLatitude(), position.getLongitude());
-                break;
-            case Event.TYPE_MAINTENANCE:
-                formatter.format(MESSAGE_TEMPLATE_TYPE_MAINTENANCE, device.getName(), position.getFixTime(),
-                        position.getLatitude(), position.getLongitude());
-                break;
-            default:
-                formatter.format("Unknown type");
-                break;
+        VelocityContext velocityContext = new VelocityContext();
+        velocityContext.put("device", device);
+        velocityContext.put("event", event);
+        if (position != null) {
+            velocityContext.put("position", position);
+            velocityContext.put("speedUnits", ReportUtils.getSpeedUnit(userId));
+        }
+        if (event.getGeofenceId() != 0) {
+            velocityContext.put("geofence", Context.getGeofenceManager().getGeofence(event.getGeofenceId()));
         }
-        String result = formatter.toString();
-        formatter.close();
-        return result;
-    }
 
-    private static String formatSpeed(long userId, double speed) {
-        DecimalFormat df = new DecimalFormat("#.##");
-        String result = df.format(speed) + " kn";
-        switch (Context.getPermissionsManager().getUser(userId).getSpeedUnit()) {
-            case "kmh":
-                result = df.format(UnitsConverter.kphFromKnots(speed)) + " km/h";
-                break;
-            case "mph":
-                result = df.format(UnitsConverter.mphFromKnots(speed)) + " mph";
-                break;
-            default:
-                break;
+        Template template = null;
+        try {
+            template = Context.getVelocityEngine().getTemplate(event.getType() + ".vm");
+        } catch (ResourceNotFoundException error) {
+            Log.warning(error);
         }
-        return result;
+        if (template == null) {
+            template = Context.getVelocityEngine().getTemplate("unknown.vm");
```

Comment:
I think we can just move this to catch. Or is there a chance that `getTemplate` returns null?